query-engine/core/src/query_document/parser.rs

```diff
@@ -152,25 +152,21 @@ impl QueryDocumentParser {
         given_arguments: &[(String, ArgumentValue)],
         query_schema: &'a QuerySchema,
     ) -> QueryParserResult<Vec<ParsedArgument<'a>>> {
-        let arguments: IndexMap<String, InputField<'_>> = schema_field
-            .arguments()
-            .map(|arg| (arg.name.clone().into_owned(), arg.clone()))
-            .collect();
-
         for (name, _) in given_arguments {
-            if !arguments.contains_key(name) {
+            if !schema_field.arguments().iter().any(|arg| arg.name == name.as_str()) {
```

Comment:
Would it be worth it to extract `schema_field.arguments()` to a variable in this function since it's called three times and can be shared, or it doesn't matter?